app/code/Academy/ProductFeed/Controller/Adminhtml/Manage/Save.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Academy\ProductFeed\Controller\Adminhtml\Manage;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+
+class Save extends Action
+{
+    public $feedFactory;
+
+    public function __construct(
+        Context                                       $context,
+        \Academy\ProductFeed\Model\ProductFeedFactory $feedFactory
+    )
+    {
+        $this->feedFactory = $feedFactory;
+        parent::__construct($context);
+    }
+
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getParams();
+
+        if (isset($data['product_feed_id']) && $data['product_feed_id']) {
+            $model = $this->feedFactory->create()->load($data['product_feed_id']);
+            $model->setFilename($data['filename'])
+                ->setName($data['name'])
+                ->setStatus($data['status'])
+                ->setFileType($data['file_type'])
+                ->setFtp($data['ftp'])
+                ->setTemplateContent($data['template_content'])
+                ->setCategories(implode(",", $data['categories']))
+                ->setUpdatedAt(date("Y-m-d H:i:s"))
+                ->save();
+            $this->messageManager->addSuccess(__('You have updated the product feed successfully.'));
+        } else {
+            $model = $this->feedFactory->create();
+            $model->setFilename($data['filename'])
```

Comment:
żeby nie powielać kodu, możesz w tym else zostawić tylko `$model->setCreatedAt()`, a resztę przenieś pod if
```
        if (isset($data['product_feed_id']) && $data['product_feed_id']) {
            $model = $this->feedFactory->create()->load($data['product_feed_id']);
            $this->messageManager->addSuccess(__('You have updated the product feed successfully.'));
        } else {
            $model = $this->feedFactory->create();
            $model->setCreatedAt(date("Y-m-d H:i:s"));
            $this->messageManager->addSuccess(__('You have successfully created the product feed.'));
        }
        $model->setFilename($data['filename'])
            ->setName($data['name'])
            ->setStatus($data['status'])
            ->setFileType($data['file_type'])
            ->setFtp($data['ftp'])
            ->setTemplateContent($data['template_content'])
            ->setCategories(implode(",", $data['categories']))
            ->setUpdatedAt(date("Y-m-d H:i:s"))
            ->save();
```

Reply:
Faktycznie powielanie jest niepotrzebne ale ta reszta o której mówisz musi być NAD if'em. Bo co jak w momencie save coś się wywali? A komunikat o sukcesie poszedł już wcześniej?

Reply:
Można przechowywać odpowiedni komunikat w zmiennej a dopiero potem po zapisie go wyświetlać.

`        if (isset($data['product_feed_id']) && $data['product_feed_id']) {
            $model = $this->feedFactory->create()->load($data['product_feed_id']);
	    $message = __('You have updated the product feed successfully.');
        } else {
            $model = $this->feedFactory->create();
            $model->setCreatedAt(date("Y-m-d H:i:s"));
	    $message = __('You have successfully created the product feed.');
            
        }
        $model->setFilename($data['filename'])
            ->setName($data['name'])
            ->setStatus($data['status'])
            ->setFileType($data['file_type'])
            ->setFtp($data['ftp'])
            ->setTemplateContent($data['template_content'])
            ->setCategories(implode(",", $data['categories']))
            ->setUpdatedAt(date("Y-m-d H:i:s"))
            ->save();
	$this->messageManager->addSuccess($message);`